scandium-core/src/main/java/org/eclipse/californium/scandium/dtls/Handshaker.java

```diff
@@ -984,6 +991,7 @@ protected final void handshakeStarted() throws HandshakeException {
 	}
 
 	protected final void sessionEstablished() throws HandshakeException {
+		amendPeerPrincipal();
```

Comment:
I assume, that `amendPeerPrincipal` should only be called once per handshake. To ensure that, consider to move it after

```
if (!sessionEstablished) {
	sessionEstablished = true;
```

Reply:
just out of curiosity: under what circumstances will *sessionEstablished* be already be `true` when this method is invoked?

Reply:
Your right. It was introduced to ensure, that this doesn't happen accidentally (may be in the future).
May be we convert that into an exception, but after 2.0.0-M16 .

Reply:
Does this mean that I should keep the method invocation at its current position?

Reply:
If you want, you could keep it there. 
I would still prefer to move it.

Reply:
ok, will do